packages/x-date-pickers-pro/src/DateRangePickerDay/DateRangePickerDay.tsx

```diff
@@ -64,20 +70,29 @@ const useUtilityClasses = (ownerState: OwnerState) => {
     isPreviewing,
     isStartOfPreviewing,
     isEndOfPreviewing,
+    isFirstVisibleCell,
+    isLastVisibleCell,
+    isHiddenDayFiller,
     selected,
     classes,
   } = ownerState;
 
   const slots = {
     root: [
       'root',
-      isHighlighting && !outsideCurrentMonth && 'rangeIntervalDayHighlight',
-      (isStartOfHighlighting || isStartOfMonth) && 'rangeIntervalDayHighlightStart',
-      (isEndOfHighlighting || isEndOfMonth) && 'rangeIntervalDayHighlightEnd',
+      isHighlighting && 'rangeIntervalDayHighlight',
+      isStartOfHighlighting && 'rangeIntervalDayHighlightStart',
+      isEndOfHighlighting && 'rangeIntervalDayHighlightEnd',
+      outsideCurrentMonth && 'rangeIntervalOutsideCurrentMonth',
+      isStartOfMonth && 'rangeIntervalStartOfMonth',
+      isEndOfMonth && 'rangeIntervalEndOfMonth',
+      isFirstVisibleCell && 'rangeIntervalFirstVisibleCell',
+      isLastVisibleCell && 'rangeIntervalLastVisibleCell',
+      isHiddenDayFiller && 'rangeIntervalHiddenDayFiller',
```

Comment:
We should add explicit consumption of these classes allowing direct "slot" override without doing root: 
```js
{[`.&${someClass}`]: { ...rules } }
```
Enabling something like to work:
```js
MuiDateRangePickerDay: {
  styleOverrides: {
    // works
    rangeIntervalDayHighlightStart: {
      backgroundColor: 'olive',
    },
    // doesn't work
    rangeIntervalLastVisibleCell: {
      backgroundColor: 'green',
    },
  },
},
```